Sources/web3swift/Transaction/AbstractEnvelope.swift

```diff
@@ -0,0 +1,133 @@
+// Package: web3swift
+// Created by Alex Vlasov.
+// Copyright © 2018 Alex Vlasov. All rights reserved.
+//
+// Support for EIP-2718 by Mark Loit 2022
+
+import Foundation
+import BigInt
+
+/// Enumeration for supported transaction types
+/// .legacy for untyped and type 0 transactions EIP155 and older
+/// .eip2930 for type 1 transactions conforming to EIP2930
+/// .eip1559 for type 2 transactions conforming to EIP1559
+public enum TransactionType: UInt, CustomStringConvertible {
+    case legacy
+    case eip2930
+    case eip1559
+
+    case unknown // catch-all type, always place as last position, also used for range checking
```

Comment:
Suggest updating to:

```
/// "catch-all" type, also used for range checking and
/// used in parsing. No TransactionObject will ever return this.
/// Must have the highest value amongst all of the `TransactionType` cases.
case unknown = UInt.max
```

Reply:
`always place as last position` got me thinking for a while until I saw this:
```
if typeUInt < TransactionType.unknown.rawValue { ...
```

To avoid that unclear comment it will be easier to just set the highest possible value for `case unknown`. It's in the code snippet of the comment above ^.

Reply:
I can try fixing up the wording, but disagree with setting it to some fixed value. The reason being is that the moment you add a new type, you will need to find and seek out any hard-coded values for what the maximum should be. Leaving room for bugs to be introduced. The way it is now it automatically tracks with the number of valid types.  Having said that, I could do something like:

```
public enum TransactionType: UInt, CustomStringConvertible {
    case legacy
    case eip2930
    case eip1559

    case total // always place as last position, used for range checking for valid input
    case unknown = UInt.max // catch-all invalid type
    ...
```

and then

```
if typeUInt < TransactionType.total.rawValue { ...

```